email/exporter_test.go

```diff
@@ -0,0 +1,123 @@
+package email
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	emailpb "github.com/letsencrypt/boulder/email/proto"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/test"
+)
+
+var ctx = context.Background()
+
+// MockPardotClientImpl is a mock implementation of PardotClient.
+type MockPardotClientImpl struct {
+	sync.Mutex
+	CreatedContacts []string
+}
+
+// NewMockPardotClientImpl returns a MockPardotClientImpl, implementing the
+// PardotClient interface. Both refer to the same instance, with the interface
+// for mock interaction and the struct for state inspection and modification.
+func NewMockPardotClientImpl() (PardotClient, *MockPardotClientImpl) {
+	mockImpl := &MockPardotClientImpl{
+		CreatedContacts: []string{},
+	}
+	return mockImpl, mockImpl
+}
+
+// SendContact adds an email to CreatedContacts.
+func (m *MockPardotClientImpl) SendContact(email string) error {
+	m.Lock()
+	defer m.Unlock()
+
+	m.CreatedContacts = append(m.CreatedContacts, email)
+	return nil
+}
+
+func (m *MockPardotClientImpl) getCreatedContacts() []string {
+	m.Lock()
+	defer m.Unlock()
+	// Return a copy to avoid race conditions.
+	return append([]string(nil), m.CreatedContacts...)
+}
+
+func setup() (*ExporterImpl, *MockPardotClientImpl, func(), func()) {
```

Comment:
Document what the two functions this returns are, and when tests should use them.